templates/starter/bicep/README.md

```diff
@@ -7,3 +7,65 @@ The following assets have been provided:
 - Infrastructure-as-code (IaC) files under the `infra` folder that demonstrate how to provision resources and setup resource tagging for `azd`.
 - A [dev container](https://containers.dev) configuration file under the `.devcontainer` directory that installs infrastructure tooling by default. This can be readily used to create cloud-hosted developer environments such as [GitHub Codespaces](https://aka.ms/codespaces).
 - Continuous deployment workflows for CI providers such as GitHub Actions under the `.github` directory, and Azure Pipelines under the `.azdo` directory that work for most use-cases.
+
+## Next Steps
+
+### Step 1: Add application code
+
+Initialize the service source code projects anywhere under the current directory. Ensure that all source code projects can be built successfully.
+
+> Note: For `function` services, it is recommended to initialize the project using the provided [quickstart tools](https://learn.microsoft.com/en-us/azure/azure-functions/functions-get-started).
+
+Once all service source code projects are building correctly, update `azure.yaml` to reference the source code projects.
+
+Run `azd package` to validate that all service source code projects can be built and packaged locally.
+
+### Step 2: Provision Azure resources
+
+Update or add Infrastructure as Code (IaC) files to provision the relevant Azure resources. This can be done incrementally, as the list of [Azure resources](https://learn.microsoft.com/en-us/azure/?product=popular) are explored and added. As an example, a reference library that contains all of the Bicep modules used by the azd templates can be found [here](https://github.com/Azure-Samples/todo-nodejs-mongo/tree/main/infra/core). All Azure resources available in Bicep format can be found [here](https://learn.microsoft.com/en-us/azure/templates/).
+
+Run `azd provision` whenever you want to ensure that changes made are applied correctly and work as expected.
+
+### Step 3: Tie in application and infrastructure
+
+There are certain pieces required that help tie-in application and infrastructure together. These are pieces that help expose your application to other Azure resources, or help allow your application to take advantage of Azure capabilities. For more details, see [below](#additional-details). This includes the following:
+
+1. Set up application settings for the code running in Azure to reference other Azure resources.
+1. Set up managed identities to allow the code running in Azure to access other Azure resources.
```

Comment:
Can we link to the section below around managed identities ( I think this is possible with some special syntax)? We start to introduce concepts without defining them.